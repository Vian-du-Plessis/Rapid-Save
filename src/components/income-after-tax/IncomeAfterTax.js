import React, {useState, useEffect} from 'react';
import styles from './IncomeAfterTax.module.scss';

const IncomeAfterTax = (props) => {

    const [value, setValue] = useState([]);

    useEffect(() => {
        setValue(props.value);
    }, [props.value])

    return (
        <div className={ styles.container }>
            <h3>{ props.title }</h3>
            <hr />
            {
                value.map((item, index) => 
                    <div key={index}>
                        <h4><strong>{item.name}</strong></h4>
                        <h5>{item.messageTwo}</h5>
                    </div>
                )
            }
        </div>
    );
};

export default IncomeAfterTax;