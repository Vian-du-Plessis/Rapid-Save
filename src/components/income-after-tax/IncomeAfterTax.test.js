import IncomeAfterTax from "./IncomeAfterTax";
import { render, screen } from "@testing-library/react";
import userEvent from '@testing-library/user-event';
import "@testing-library/jest-dom/extend-expect";
import renderer from 'react-test-renderer';

const object = [{
    name: 'Vian',
    messageTwo: 'Vian paid R10385 to tax and has a net income of R138559 for the year'
}]

const IncomeAfterTaxProps = {
    title: 'Income after Tax',
    value: object
}

describe("Test Income After Tax Component...", () => {

    test("Show correct title...", () => {
        render(
            <IncomeAfterTax
                {...IncomeAfterTaxProps}
            />
        );
        const titleElement = screen.getByText("Income after Tax");

        expect(titleElement).toBeInTheDocument();
        expect(titleElement.textContent).toBe("Income after Tax");
    });

    test("Show correct sub Heading One...", () => {
        render(
            <IncomeAfterTax
                {...IncomeAfterTaxProps}
            />
        );
        const titleElement = screen.getByText("Vian paid R10385 to tax and has a net income of R138559 for the year");

        expect(titleElement).toBeInTheDocument();
        expect(titleElement.textContent).toBe("Vian paid R10385 to tax and has a net income of R138559 for the year");
    });

    // SNAPSHOT TEST
    test("Snapshot of component", () => {
        const component = renderer.create(
            <IncomeAfterTax/>
        );

        let tree = component.toJSON();

        expect(tree).toMatchSnapshot();

        tree = component.toJSON();
        expect(tree).toMatchSnapshot()
    })

})