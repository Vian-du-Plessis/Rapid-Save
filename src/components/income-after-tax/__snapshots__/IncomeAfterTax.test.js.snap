// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Income After Tax Component...  Snapshot of component 1`] = `
<div
  className="container"
>
  <h3 />
  <hr />
</div>
`;

exports[`Test Income After Tax Component...  Snapshot of component 2`] = `
<div
  className="container"
>
  <h3 />
  <hr />
</div>
`;
